great_expectations/checkpoint/actions.py

```diff
@@ -869,71 +870,51 @@ def _run(  # type: ignore[override] # signature does not match parent  # noqa: P
-            validation_result_suite_identifier,
-            (ValidationResultIdentifier, GXCloudIdentifier),
-        ):
-            raise TypeError(
-                "validation_result_id must be of type ValidationResultIdentifier or GeCloudIdentifier, not {}.".format(
-                    type(validation_result_suite_identifier)
-                )
-            )
-
-        checkpoint_id = None
-        if self._using_cloud_context and checkpoint_identifier:
-            checkpoint_id = checkpoint_identifier.id
-
-        expectation_suite_id = None
-        if self._using_cloud_context and expectation_suite_identifier:
-            expectation_suite_id = expectation_suite_identifier.id
-
-        return self.target_store.set(
-            validation_result_suite_identifier,
-            validation_result_suite,
-            checkpoint_id=checkpoint_id,
-            expectation_suite_id=expectation_suite_id,
-        )
-
-    def _run_cloud_post_process_resource_ref(
-        self,
-        gx_cloud_resource_ref: GXCloudResourceRef,
-        validation_result_suite_identifier: Union[
-            ValidationResultIdentifier, GXCloudIdentifier
-        ],
+def store_validation_results(
+    validation_result_store: ValidationsStore,
+    suite_validation_result: ExpectationSuiteValidationResult,
+    suite_validation_result_identifier: ValidationResultIdentifier | GXCloudIdentifier,
+    expectation_suite_identifier: Optional[
+        ExpectationSuiteIdentifier | GXCloudIdentifier
+    ] = None,
+    checkpoint_identifier: Optional[GXCloudIdentifier] = None,
+) -> Any:
+    """Helper function to do the heavy lifting for StoreValidationResultAction and ValidationConfigs.
+    This is broken from the ValidationAction (for now) so we don't need to pass the data_context around.
+    """
+    using_cloud_context = validation_result_store.cloud_mode
+    checkpoint_id = None
+    if using_cloud_context and checkpoint_identifier:
+        checkpoint_id = checkpoint_identifier.id
+
+    expectation_suite_id: Optional[str] = None
+    if isinstance(expectation_suite_identifier, GXCloudIdentifier):
+        expectation_suite_id = expectation_suite_identifier.id
+
+    run_return_value = validation_result_store.set(
+        key=suite_validation_result_identifier,
+        value=suite_validation_result,
+        checkpoint_id=checkpoint_id,
+        expectation_suite_id=expectation_suite_id,
+    )
+    if (
+        using_cloud_context
+        and isinstance(run_return_value, GXCloudResourceRef)
+        and isinstance(suite_validation_result_identifier, GXCloudIdentifier)
     ):
-        store_set_return_value: GXCloudResourceRef
-        new_id = gx_cloud_resource_ref.id
-        # ValidationResultIdentifier has no `.id`
-        validation_result_suite_identifier.id = new_id  # type: ignore[union-attr]
-        return gx_cloud_resource_ref
+        new_id = suite_validation_result_identifier.id
+        assert new_id is not None
```

Comment:
Instead of a raw assert could we raise an Exception with a descriptive error message?